react/jest.config.cjs

```diff
@@ -28,9 +28,7 @@ module.exports = {
   coverageDirectory: 'coverage',
 
   // An array of regexp pattern strings used to skip coverage collection
-  coveragePathIgnorePatterns: [
-    "src/core-components/"
-  ],
+  coveragePathIgnorePatterns: ['src/core-components/', 'src/core-wrappers'],
```

Comment:
I ran `npm ci` and `npm run test` with this new `coveragePathIgnorePatterns`, and with an empty list. Both produced the same results. Does that mean tests are passing for `src/core-…`? Should tests then be allowed ot run? Am I misunderstanding?

<details><summary><code>  coveragePathIgnorePatterns: ['src/core-components/', 'src/core-wrappers'],</code></summary>

```
...
----------------------------|---------|----------|---------|---------|-------------------
Test Suites: 3 failed, 4 passed, 7 total
Tests:       4 passed, 4 total
Snapshots:   0 total
Time:        16.494 s
Ran all test suites.
```

</details>
<details><summary><code>  coveragePathIgnorePatterns: [],</code></summary>

```
...
----------------|---------|----------|---------|---------|-------------------
Test Suites: 3 failed, 4 passed, 7 total
Tests:       4 passed, 4 total
Snapshots:   0 total
Time:        14.151 s
Ran all test suites.
```

</details>

Reply:
One can argue "ignore lint, cuz it is responsibility of future package; thus ignore tests for the same reason". But if the tests pass, this "ignore" code seems destined to become cruft.

Reply:
coveragePathIgnorePatterns should be related to the coverage checks for those dir, while testPathIgnorePatterns is ignoring the actual tests for those directories. Those 7 tests suites and 4 tests you got with npm run test are all outside those directories, so I'm surprised 3 test suites failed for you. Don't think that happened for Nathan and me. Hmm. Might try removing node modules and the package-lock and running npm install and then run the tests again and see if those 3 still fail?

Reply:
Thanks. Heh, I had not distinguished coverage and test. 😅

Note: I ran —

1. `cd react`
2. `rm -rf node_modules`
3. `npm ci`
4. `npm run test`

— then —

1. `rm -rf node_modules`
2. `rm package-lock.json`
3. `npm install`
4. `npm run test`

<details><summary>Same output each time.</summary>

```
Test Suites: 3 failed, 4 passed, 7 total
Tests:       4 passed, 4 total
Snapshots:   0 total
```

</details>

Reply:
When you get the chance, do you mind sharing which 3 are failing for you and any error output those 3 tests produce that could hint at the issue

Reply:
Sure.

<details><summary>Output</summary>

```
➜  react git:(master) npm run test

> hazmapper@0.0.0 test
> jest


 RUNS  src/pages/Callback/Callback.test.tsx
 RUNS  src/components/Map/Map.test.tsx

 RUNS  src/pages/Callback/Callback.test.tsx
 RUNS  src/components/Map/Map.test.tsx

 RUNS  src/pages/Callback/Callback.test.tsx
 RUNS  src/components/Map/Map.test.tsx

 RUNS  src/pages/Callback/Callback.test.tsx
 RUNS  src/components/Map/Map.test.tsx

 RUNS  src/pages/Callback/Callback.test.tsx
 RUNS  src/components/Map/Map.test.tsx

 RUNS  src/pages/Callback/Callback.test.tsx
 RUNS  src/components/Map/Map.test.tsx

 RUNS  src/pages/Callback/Callback.test.tsx
 RUNS  src/components/Map/Map.test.tsx

 RUNS  src/pages/Callback/Callback.test.tsx

 RUNS  src/pages/Callback/Callback.test.tsx

 RUNS  src/pages/Callback/Callback.test.tsx

 RUNS  src/pages/Callback/Callback.test.tsx

 RUNS  src/pages/Callback/Callback.test.tsx

 RUNS  src/pages/Callback/Callback.test.tsx
  console.warn
    Warning: `fetch` is not available. Please supply a custom `fetchFn` property to use `fetchBaseQuery` on SSR environments.

       9 | export const geoapi = createApi({
      10 |   reducerPath: 'api',
    > 11 |   baseQuery: fetchBaseQuery({
         |                            ^
      12 |     baseUrl: BASE_URL,
      13 |     prepareHeaders: (headers, api) => {
      14 |       // TODO check if logged in as we don't want to add if public

      at fetchBaseQuery (node_modules/@reduxjs/toolkit/dist/query/rtk-query.cjs.development.js:363:13)
      at Object.<anonymous> (src/redux/api/geoapi.ts:11:28)
      at Object.require (src/redux/reducers/reducers.ts:2:1)
      at Object.require (src/redux/store.ts:2:1)
      at Object.require (src/requests.ts:2:1)
      at Object.<anonymous> (src/requests.test.ts:1:1)


 RUNS  src/pages/Callback/Callback.test.tsx
  console.warn
    Warning: `fetch` is not available. Please supply a custom `fetchFn` property to use `fetchBaseQuery` on SSR environments.

       9 | export const geoapi = createApi({
      10 |   reducerPath: 'api',
    > 11 |   baseQuery: fetchBaseQuery({
         |                            ^
      12 |     baseUrl: BASE_URL,
      13 |     prepareHeaders: (headers, api) => {
      14 |       // TODO check if logged in as we don't want to add if public

      at fetchBaseQuery (node_modules/@reduxjs/toolkit/dist/query/rtk-query.cjs.development.js:363:13)
      at Object.<anonymous> (src/redux/api/geoapi.ts:11:28)
      at Object.require (src/redux/reducers/reducers.ts:2:1)
      at Object.require (src/redux/store.ts:2:1)
      at Object.require (src/requests.ts:2:1)
      at Object.require (src/hooks/projects/useProjects.ts:3:1)
      at Object.require (src/hooks/index.ts:1:1)
      at Object.require (src/pages/MainMenu/MainMenu.tsx:8:1)
      at Object.<anonymous> (src/pages/MainMenu/MainMenu.test.tsx:3:1)


 RUNS  src/pages/Callback/Callback.test.tsx

 RUNS  src/pages/Callback/Callback.test.tsx
  console.warn
    Warning: `fetch` is not available. Please supply a custom `fetchFn` property to use `fetchBaseQuery` on SSR environments.

       9 | export const geoapi = createApi({
      10 |   reducerPath: 'api',
    > 11 |   baseQuery: fetchBaseQuery({
         |                            ^
      12 |     baseUrl: BASE_URL,
      13 |     prepareHeaders: (headers, api) => {
      14 |       // TODO check if logged in as we don't want to add if public

      at fetchBaseQuery (node_modules/@reduxjs/toolkit/dist/query/rtk-query.cjs.development.js:363:13)
      at Object.<anonymous> (src/redux/api/geoapi.ts:11:28)
      at Object.require (src/redux/reducers/reducers.ts:2:1)
      at Object.require (src/redux/store.ts:2:1)
      at Object.<anonymous> (src/pages/Callback/Callback.test.tsx:5:1)


 RUNS  src/pages/Callback/Callback.test.tsx
 PASS  src/pages/StreetviewCallback/StreetviewCallback.test.tsx (12.653 s)

 RUNS  src/pages/Callback/Callback.test.tsx
  ✓ renders streetview callback (63 ms)

 RUNS  src/pages/Callback/Callback.test.tsx


 RUNS  src/pages/Callback/Callback.test.tsx
 PASS  src/pages/Callback/Callback.test.tsx (12.72 s)

 RUNS  src/pages/Callback/Callback.test.tsx
  ✓ renders callback (69 ms)

 RUNS  src/pages/Callback/Callback.test.tsx


 RUNS  src/pages/Callback/Callback.test.tsx
  console.warn
    Warning: `fetch` is not available. Please supply a custom `fetchFn` property to use `fetchBaseQuery` on SSR environments.

       9 | export const geoapi = createApi({
      10 |   reducerPath: 'api',
    > 11 |   baseQuery: fetchBaseQuery({
         |                            ^
      12 |     baseUrl: BASE_URL,
      13 |     prepareHeaders: (headers, api) => {
      14 |       // TODO check if logged in as we don't want to add if public

      at fetchBaseQuery (node_modules/@reduxjs/toolkit/dist/query/rtk-query.cjs.development.js:363:13)
      at Object.<anonymous> (src/redux/api/geoapi.ts:11:28)
      at Object.require (src/redux/reducers/reducers.ts:2:1)
      at Object.require (src/redux/store.ts:2:1)
      at Object.require (src/requests.ts:2:1)
      at Object.require (src/hooks/projects/useProjects.ts:3:1)
      at Object.require (src/hooks/index.ts:1:1)
      at Object.require (src/pages/Login/Login.tsx:6:1)
      at Object.<anonymous> (src/pages/Login/Login.test.tsx:3:1)

 PASS  src/components/Map/Map.test.tsx (12.921 s)
  ✓ renders map (424 ms)

  console.warn
    Warning: `fetch` is not available. Please supply a custom `fetchFn` property to use `fetchBaseQuery` on SSR environments.

       9 | export const geoapi = createApi({
      10 |   reducerPath: 'api',
    > 11 |   baseQuery: fetchBaseQuery({
         |                            ^
      12 |     baseUrl: BASE_URL,
      13 |     prepareHeaders: (headers, api) => {
      14 |       // TODO check if logged in as we don't want to add if public

      at fetchBaseQuery (node_modules/@reduxjs/toolkit/dist/query/rtk-query.cjs.development.js:363:13)
      at Object.<anonymous> (src/redux/api/geoapi.ts:11:28)
      at Object.require (src/redux/authSlice.ts:8:1)
      at Object.require (src/pages/Logout/Logout.tsx:4:1)
      at Object.<anonymous> (src/pages/Logout/Logout.test.tsx:3:1)

  console.warn
    Detected an action creator with type "auth/logout" being dispatched. 
    Make sure you're calling the action creator before dispatching, i.e. `dispatch(logout())` instead of `dispatch(logout)`. This is necessary even if the action has no payload.

      13 |
      14 |   useEffect(() => {
    > 15 |     dispatch(logout);
         |     ^
      16 |   }, [dispatch]);
      17 |
      18 |   return (

      at node_modules/@reduxjs/toolkit/src/immutableStateInvariantMiddleware.ts:160:31
      at dispatch (src/pages/Logout/Logout.tsx:15:5)
      at commitHookEffectListMount (node_modules/react-dom/cjs/react-dom.development.js:3926:112)
      at commitPassiveMountOnFiber (node_modules/react-dom/cjs/react-dom.development.js:4139:1573)
      at commitPassiveMountEffects_complete (node_modules/react-dom/cjs/react-dom.development.js:4139:869)
      at commitPassiveMountEffects_begin (node_modules/react-dom/cjs/react-dom.development.js:4139:570)
      at commitPassiveMountEffects (node_modules/react-dom/cjs/react-dom.development.js:4139:196)
      at flushPassiveEffectsImpl (node_modules/react-dom/cjs/react-dom.development.js:4636:402)
      at flushPassiveEffects (node_modules/react-dom/cjs/react-dom.development.js:4631:371)
      at node_modules/react-dom/cjs/react-dom.development.js:4566:84
      at flushActQueue (node_modules/react/cjs/react.development.js:2332:28)
      at act (node_modules/react/cjs/react.development.js:2256:15)
      at node_modules/@testing-library/react/dist/act-compat.js:87:25
      at renderRoot (node_modules/@testing-library/react/dist/pure.js:171:26)
      at render (node_modules/@testing-library/react/dist/pure.js:256:10)
      at Object.<anonymous> (src/pages/Logout/Logout.test.tsx:9:31)

 PASS  src/pages/Logout/Logout.test.tsx (13.269 s)
  ✓ renders logout (283 ms)

 FAIL  src/requests.test.ts
  ● Test suite failed to run

    src/hooks/environment/useAppConfiguration.ts:4:47 - error TS2307: Cannot find module '../../secret_local' or its corresponding type declarations.

    4 import { localDevelopmentConfiguration } from '../../secret_local';
                                                    ~~~~~~~~~~~~~~~~~~~~

 FAIL  src/pages/MainMenu/MainMenu.test.tsx
  ● Test suite failed to run

    src/hooks/environment/useAppConfiguration.ts:4:47 - error TS2307: Cannot find module '../../secret_local' or its corresponding type declarations.

    4 import { localDevelopmentConfiguration } from '../../secret_local';
                                                    ~~~~~~~~~~~~~~~~~~~~

 FAIL  src/pages/Login/Login.test.tsx
  ● Test suite failed to run

    src/hooks/environment/useAppConfiguration.ts:4:47 - error TS2307: Cannot find module '../../secret_local' or its corresponding type declarations.

    4 import { localDevelopmentConfiguration } from '../../secret_local';
                                                    ~~~~~~~~~~~~~~~~~~~~

----------------------------|---------|----------|---------|---------|-------------------
File                        | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
----------------------------|---------|----------|---------|---------|-------------------
All files                   |   70.73 |    27.27 |   51.72 |   70.73 |                   
 __fixtures__               |     100 |      100 |     100 |     100 |                   
  featuresFixture.ts        |     100 |      100 |     100 |     100 |                   
  tileServerLayerFixture.ts |     100 |      100 |     100 |     100 |                   
 components/Map             |   88.88 |      100 |   71.42 |   88.88 |                   
  Map.tsx                   |   88.88 |      100 |   71.42 |   88.88 | 30,70             
 pages/Callback             |      75 |       50 |     100 |      75 |                   
  Callback.tsx              |      75 |       50 |     100 |      75 | 19-23,31-34       
 pages/Logout               |   91.66 |      100 |   66.66 |   91.66 |                   
  Logout.tsx                |   91.66 |      100 |   66.66 |   91.66 | 11                
 pages/StreetviewCallback   |     100 |      100 |     100 |     100 |                   
  StreetviewCallback.tsx    |     100 |      100 |     100 |     100 |                   
 redux                      |   67.85 |      100 |    37.5 |   67.85 |                   
  authSlice.ts              |      50 |      100 |      20 |      50 | 25-41,48-52       
  projectsSlice.ts          |   83.33 |      100 |      50 |   83.33 | 13                
  store.ts                  |     100 |      100 |     100 |     100 |                   
 redux/api                  |   41.66 |        0 |      25 |   41.66 |                   
  geoapi.ts                 |   41.66 |        0 |      25 |   41.66 | 15-27,33-37       
 redux/reducers             |     100 |      100 |     100 |     100 |                   
  reducers.ts               |     100 |      100 |     100 |     100 |                   
 utils                      |      45 |        0 |      25 |      45 |                   
  authUtils.ts              |      45 |        0 |      25 |      45 | 6-14,22-26,32,36  
----------------------------|---------|----------|---------|---------|-------------------
Test Suites: 3 failed, 4 passed, 7 total
Tests:       4 passed, 4 total
Snapshots:   0 total
Time:        16.61 s
Ran all test suites.
```

</details>

_P.S. The `package-lock.json` did not change upon removal and `npm install`, so I think it is not the culprit._

Reply:
Ah I see the issue. To run locally with our project you need to create a secret_local.ts file (it gets ignored by git since it's got a secret JWT in it). Then put your JWT in it. We have an example in the project called secrets_local.example.ts. So the tests are failing for you bc my useAuthenticated hook needs an authenticated user to work.

Reply:
Done. Tests pass. Thanks.